File: API/WebsiteApi/Models/Job.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebsiteApi.Models
{
    public class Job
    {
        [Key]
        public int JobId { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; }
        public required string CarModel { get; set; }

        [EnumDataType(typeof(Status))]
        public Status JobStatus { get; set; }

        public string? Supervisor { get; set; }

        [Column(TypeName = "decimal(10,2)")]
        public Decimal Price { get; set; }
        public int LaborHours { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime? UpdatedAt { get; set; }
        public ICollection<Expense> Expenses { get; set; }

        public Job()
        {
            this.JobStatus = Status.Pending;
            this.LaborHours = 0;
        }
    }

    public enum Status
    {
        Pending,
        InProgress,
        Completed,
        Cancelled
    }
}

File: WebApp/models/Job.ts
// Job interface matching backend model
interface Job {
    jobId: number;
    name: string;
    description?: string;
    carModel: string;
    jobStatus: Status;
    supervisor?: string;
    price: number;
    createdAt: Date;
    updatedAt?: Date;
}

// Enum for Job Status
enum Status {
    Pending = 0,
    InProgress = 1,
    Completed = 2,
    Cancelled = 3,
}

// JobFilters interface for filters state
interface IJobFilters {
    jobId?: number;
    name?: string;
    description?: string;
    carModel?: string;
    jobStatus?: Status;
    supervisor?: string;
    price?: number;
  }